ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/acs/ActiveContractSetSource.scala

```diff
@@ -21,10 +21,11 @@ object ActiveContractSetSource {
   def apply(
       stub: (GetActiveContractsRequest, StreamObserver[GetActiveContractsResponse]) => Unit,
       request: GetActiveContractsRequest)(implicit esf: ExecutionSequencerFactory)
-    : Source[GetActiveContractsResponse, Future[String]] = {
+    : Source[GetActiveContractsResponse, Future[Option[String]]] = {
 
     ClientAdapter
       .serverStreaming(request, stub)
-      .viaMat(new ExtractMaterializedValue(r => Some(r.offset)))(Keep.right)
+      .viaMat(new ExtractLastMaterializedValue(r =>
+        if (r.offset.nonEmpty) Some(r.offset) else None))(Keep.right)
```

Comment:
When should I expect this to return `None`? The docs in the proto state that the offset is required so I’m not sure when it is empty.

Reply:
The materialized value will be None if the stream terminates with an error. Alternatively we could also fail the materialized future.

Reply:
Failing seems more intuitive to me but I’m happy to leave this decision to you. It would however be nice to have a comment on `getActiveContracts` that explains this.

Reply:
Thanks for the input, @cocreature. I went back to failing the materialized future.